### ISSUE TITLE
Black bars flash for a second when pushing back.

### ISSUE THREAD

**Issue by AndyIbanez** [younatics/MediaBrowser#13]
There seems to be a weird bug - that does not affect MediaBrowser only - in which a black bar shows up for a second when popping back from the MediaBrowser view controller. You can observe said behavior on the gifs provided by the project itself.

![Black Bars](https://github.com/younatics/MediaBrowser/raw/master/Images/SinglePhoto.gif?raw=true)

You can see the image show up for a second when the browser pushes back. At first I didn't think it was so bad, but it can be bit surprising when you actually see the bars happening on your device.

Here's a still shot of the black bar:

![Black bar](https://i.imgur.com/FHcJ4G6.png)

MWPhotoBrowser also has this problem.

**Comment by weakfl** [comment]
@AndyIbanez this is more or less an iOS issue. Setting your navbar's background image to an empty image should fix this:
```swift
navigationController?.navigationBar.setBackgroundImage(UIImage(), for: .default)
```

NOTE: You'll probably also have to set the MediaBrowser's `navigationBarBackgroundColor` property to `UIColor.clear` to make it work on push AND pop.
Take a look at the [fix-issue-4 branch](https://github.com/younatics/MediaBrowser/tree/fix-issue-4) where I've fixed this issue...

@younatics 
I didn't fully understand why you are setting an y offset to the collectionview's frame of the `MediaGridViewController` in `MediaBrowser`, line 1761:
```swift
gc.view.frame = CGRect(x: 0.0, y: naviHeight, width: bounds.width, height: bounds.height - naviHeight)
```

This will cause awkward transitions when `startOnGrid` is `true`. The content offset of the collectionview should be adjusted automatically anyway?

**Comment by weakfl** [comment]
@younatics The solution needs some refinement before it can be merged, I'll keep you posted...

**Comment by younatics** [comment]
@weakfl Thanks!

**Comment by weakfl** [comment]
@younatics I think the best solution is to restore some of the previosViewController's navigationbar properties in `willMove(toParentViewController parent: UIViewController?)`. It results in better transitions as long as the previousViewController has set a barTintColor and you don't have to change the backgroundImage...

Please take a look at 7a910b6 and let me know what you think.

**Comment by younatics** [comment]
@weakfl ,As I think, it is great idea to change code. You can merge this branch to `master`. Thanks!

**Comment by weakfl** [comment]
Merged an released, see [2.05](https://github.com/younatics/MediaBrowser/releases/tag/2.0.5)
